optas/builder.py

```diff
@@ -210,6 +212,104 @@ def get_model_states(self, name, time_deriv=0):
         return self._decision_variables[name]
 
 
+    def get_model_parameters(self, name, time_deriv=0):
+        """Get the vector of parameters for a given model.
+
+        Syntax
+        ------
+
+        parameters = builder.get_model_parameters(name, time_deriv=0)
+
+        Parameters
+        ----------
+
+        name (string)
+            Name of the model.
+
+        time_deriv (int)
+            The time-deriviative required (i.e. position is 0, velocity is 1, etc.)
+
+        Returns
+        -------
+
+        parameters (casadi.SX, with shape dim-by-T)
+            The vector of parameters where dim is the number of model parameters, and T is the number of time-steps in the trajectory.
+
+        """
+        model = self.get_model(name)
+        assert time_deriv in model.time_derivs, f"model '{name}', was not specified with time derivative to order {time_deriv}"
+        name = model.parameter_name(time_deriv)
+        return self._parameters[name]
+
+
+    def get_model_parameter(self, name, t, time_deriv=0):
+        """Get the model parameter at a given time.
+
+        Syntax
+        ------
+
+        parameter = builder.get_model_parameter(name, t, time_deriv=0)
+
+        Parameters
+        ----------
+
+        name (string)
+            Name of the model.
+
+        t (int)
+            Index of the desired parameter.
+
+        time_deriv (int)
+            The time-deriviative required (i.e. position is 0, velocity is 1, etc.)
+
+        Returns
+        -------
+
+        parameter (casadi.SX, with shape dim-by-1)
+            The parameter vector where dim is the model number of parameters
+
+        """
+        parameters = self.get_model_parameters(name, time_deriv)
+        return parameters[:, t]
+
+
+    def get_model_states_and_parameters(self, name, time_deriv=0):
+        """Get the vector of states and parameters for a given model.
+
+        Syntax
+        ------
+
+        parameters = builder.get_model_states_and_parameters(name, time_deriv=0)
+
+        Parameters
+        ----------
+
+        name (string)
+            Name of the model.
+
+        time_deriv (int)
+            The time-deriviative required (i.e. position is 0, velocity is 1, etc.)
+
+        Returns
+        -------
+
+        states and parameters (casadi.SX, with shape dim-by-T)
+            The vector of parameters where dim is the number of model states and parameters (for a robot it should correspond to the degrees of freedom), and T is the number of time-steps in the trajectory.
+
+        """
+        states = self.get_model_states(name, time_deriv=time_deriv)
+        parameters = self.get_model_parameters(name, time_deriv=time_deriv)
+
+        model = self.get_model(name)
+
+        states_and_params = cs.SX.zeros(model.dim, self.T-time_deriv)
```

Comment:
Won't this break when `optimize_time=True`?

Reply:
Don't know. I haven't tested with optimize_time=True.
How can we test?